packages/sdk/app-graph/src/action.ts

```diff
@@ -10,6 +10,10 @@ import type { MaybePromise } from '@dxos/util';
 // TODO(thure): `Parameters<TFunction>` causes typechecking issues because `TFunction` has so many signatures.
 export type Label = string | [string, { ns: string; count?: number }];
 
+export type InvokeParams = Partial<{
+  caller: string;
```

Comment:
Interesting, do you think this is generic enough that the typedef is a framework-level thing? I was envisioning that the params would be type unknown and you'd have to validate them in the invoke (sort of like surface component data). That way you could pass in anything, but if you want to use it you have to validate it is what you think it is.

Reply:
Yeah I wasn’t entirely sure about this approach. We do a lot of this validation sniffing which is logically expensive and may be unnecessary here, so I figured this should probably assume less at the outset, hence the framework-level typedef.